frontend/src/components/DataDashboard.vue

```diff
@@ -29,52 +35,42 @@ const downloadCSV = (gender) => {
     const url = window.URL.createObjectURL(blob);
     const a = document.createElement('a');
     a.setAttribute('href', url);
-    a.setAttribute('download', genderToDcid(gender) + '.csv');
+    a.setAttribute('download', `${dcid}.csv`);
     a.click();
   } catch (err) {
     console.log(err);
   }
   loading_download.value = false;
 }
-
-let apiCache = new ApiCache();
-
-async function getData(dcid: string) {
-
-  // This uses DataCommons' public API key. DO NOT INCLUDE A PRIVATE API KEY HERE!
-  let request = "https://api.datacommons.org/v1/observations/series/country/USA/" + dcid + "?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI";
-
-  if (apiCache.recordExists(request)) {
-    tableItems.value = apiCache.get(request).observations;
+ async function getData(dcid: string) {
+     const values = await dcClient.getData(dcid)
+     tableItems.value = values
   }
 
-  else {
-    const res = await fetch(request);
-    const finalRes = await res.json();
-    tableItems.value = finalRes.observations;
-    apiCache.set(request, finalRes);
-  }
-}
-
-function genderToDcid(gender: string): string {
-  let dcidMap = new Map<string, string>([
-    ["Males", "Count_Person_25OrMoreYears_EducationalAttainmentBachelorsDegree_Male"],
-    ["Females", "Count_Person_25OrMoreYears_EducationalAttainmentBachelorsDegree_Female"]
-  ]);
-
-  let dcid = dcidMap.get(gender);
-  return dcid == undefined ? "" : dcid;
-}
-
-watchEffect(async () => {
-  getData(genderToDcid(gender.value))
-}) 
+ const genderToDcid = (gender: string): string => {
+     let query: string = 'Female'
+     if (gender == 'Males') {
```

Comment:
Avoiding this logic seems like a good reason to make sure our checkbox filter strings match our query strings wherever possible for the real data sets. IMO you should feel free to take liberties tweaking the wording between mocks and implementation if necessary.

Reply:
We can make constants that map to category dimensions. example

``` 
// Within data/dimensions.ts
const Male = 'Male'
const Female = 'Female'

// Within DataDashboard.vue
import { Male, Female } from ../data/dimensions

// ...

const gender = ref(Male) 
// ...

<RadioButton id="gender" name="gender" value=Male v-model="gender" />

// ...

const genderToDcid = (gender: string): string => {
     const genderQuery = new Query('gender', gender)
     const majorQuery = new Query('major', 'BachelorOfScienceAndEngineeringMajor')

```

Does this approach seem sustainable/scalable?

Reply:
_added constants_